crates/re_viewer/src/ui/recordings_panel.rs

```diff
@@ -34,58 +40,93 @@ fn recording_list_ui(ctx: &mut ViewerContext<'_>, ui: &mut egui::Ui) {
         ..
     } = ctx;
 
-    let mut store_dbs = store_context.alternate_recordings.clone();
-    if store_dbs.is_empty() {
-        return;
+    let mut store_dbs_map: BTreeMap<_, Vec<_>> = BTreeMap::new();
+    for store_db in &store_context.alternate_recordings {
+        store_dbs_map
+            .entry(
+                store_db
+                    .store_info()
+                    .map_or("<unknown>", |info| info.application_id.as_str()),
+            )
+            .and_modify(|store_dbs| store_dbs.push(*store_db))
+            .or_insert(vec![store_db]);
```

Comment:
nit: slightly simpler

```suggestion
            .or_default()
            .push(*store_db);
```